### DISCUSSION TITLE
Fields on relations

### DISCUSSION THREAD

**Discussion by hiasl** [craftcms/cms#10246] [Ideas]
A typical customer request: Can we set / override an image's caption in an entry?
- Normally we have Copyright, Alt-Text, ... as fields on the image/asset
- When an editor chooses an image, the alt-text is taken from the asset
- To override it or set it on an entry basis things start getting ugly. E.g. creating a matrix with the asset field and a text field inside, or multi-row text fields which are split up...

Solution: 
- Allow fields to be assigned to relations (entries, assets, users, ...)
- This should happen in the field definition process, which creates the relation in the entry. This means when you create e.g. an asset field, you should be able to assign other fields to it for the relation
- These fields are then available to be filled out for each asset assigned when you edit the entry that contains the asset field.
- Fields to be assignable to relations should be the same as to entry types
- When an element to which a relation is created, contains the same field, its value would be overridden (maybe also cancelable)
- When an element to which a relation is created does not contain the field from the relation, its value is available in addition
- The GUI should stay the same as it currently is with all relations. Double clicking the relation slides in the modal from the right side. Fields from the relation are edited inside of the modal in addition to the fields from the related element.

Possible use cases:
- Image Captions with images / assets
- Project roles for users in a team listing of a project
- ... and many more

**Comment by brandonkelly** [comment]
This has actually come up before – see #8509.

**Reply by hiasl** [reply]
Thanks @brandonkelly, I did not find this.
And yes, the initial requirement was the same,  but I think my concept of putting fields on the relation by defining it on the source of the relation makes it more universal and therefor better than the suggestion in the other discussion.

**Reply by brandonkelly** [reply]
Would be great if you could post over there with whatever thoughts you have.

**Reply by hiasl** [reply]
I'll do
